griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -0,0 +1,159 @@
+import uuid
+from typing import Optional
+from attr import define, field, Factory
+from griptape.drivers import BaseVectorStoreDriver
+from sqlalchemy.engine import Engine
+from sqlalchemy import create_engine, Column, String, JSON
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Session
+from pgvector.sqlalchemy import Vector
+
+
+@define
+class PgVectorVectorStoreDriver(BaseVectorStoreDriver):
+    """
+    A vector store driver to Postgres using the PGVector extension.
+
+    Attributes:
+        connection_string: The connection string for the Postgres database instance.
+        create_engine_params: Additional configuration params passed when creating the database connection.
+        engine: An optional sqlalchemy Postgres engine to use.
+    """
+
+    connection_string: Optional[str] = field(default=None, kw_only=True)
```

Comment:
Remove `Optional`, `default`

Reply:
Will this prevent us from supporting bring-your-own-engine here? The approach I aimed for is either provide the connection string (and we'll build the engine) or provide the engine directly.

Reply:
Good point, I was looking at the [other drivers](https://github.com/griptape-ai/griptape/blob/main/griptape/drivers/vector/marqo_vector_store_driver.py#L21) as examples but I think it's actually incorrectly implemented on them. If both field are optional, maybe we should add a [validator](https://github.com/griptape-ai/griptape/blob/main/griptape/mixins/activity_mixin.py#L13) that ensures that at least one is set.